app/src/androidTest/java/com/github/se/project/ui/lesson/AddLessonTest.kt

```diff
@@ -269,19 +276,23 @@ class AddLessonTest {
 
     composeTestRule.waitForIdle()
     composeTestRule.onNodeWithTag("instantButton").performClick()
-    composeTestRule.waitForIdle()
+    composeTestRule.waitUntil(20000) { composeTestRule.onNodeWithTag("mapButton").isNotDisplayed() }
 
     // Set Subject and Language
+    composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
     composeTestRule.onNodeWithTag("subjectButton").performClick()
     composeTestRule.waitForIdle()
     composeTestRule.onNodeWithTag("dropdown${Subject.AICC}").performClick()
-    composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
 
     composeTestRule.waitForIdle()
 
+    composeTestRule.waitUntil(20000) {
+      composeTestRule.onNodeWithText(context.getString(R.string.select_subject)).isNotDisplayed()
+    }
```

Comment:
It is really good that you used waitUntil functions instead of other more hardcoded sleep, good job!